pkg/controller/cdapmaster/cdapmaster_controller.go

```diff
@@ -370,12 +458,60 @@ func (b *Base) Objects(rsrc interface{}, rsrclabels map[string]string, observed,
 	}
 	var err error
 	for k, v := range configs {
-		resources, err = addConfigMapItem(master, getConfigName(master, k), labels, v, resources)
+		resources, err = addConfigMapItem(master, getResourceName(master, k), labels, v, resources)
 		if err != nil {
 			return nil, err
 		}
 	}
 
+	var item metav1.Object
+	if master.Status.UpgradeJobVersion == "" && master.Status.ImageToUse != master.Spec.Image {
```

Comment:
You can return early if this condition is not true as there is no upgrade needed, right? I believe it can simplify the following big if-else block.